File: lib/api/image_slide/edit_image_slide.dart
import 'package:ut_report_generator/api/send_request.dart';
import 'package:ut_report_generator/models/response/edit_image_slide_response.dart';

Future<EditImageSlideResponse> editSlide({
  required String slideshow,
  required String imageSlide,
  required String parameterName,
  required String parameterValue,
}) {
  return sendRequest(
    route: "image_slide/edit",
    body: {
      "report": slideshow,
      "image_slide": imageSlide,
      "parameter_name": parameterName,
      "parameter_value": parameterValue,
    },
    callback: EditImageSlideResponse.fromJson,
  );
}

File: python-app/routes/image_slide/edit_image_slide.py
from lib.with_flask import with_flask
from lib.get_or_panic import get_or_panic
from lib.get_entities_from_request import entities_for_editing_image_slide
from lib.slide.render_preview import render_preview
from lib.slide.delete_preview import delete_preview
from lib.report.save_slideshow import save_slideshow

from models.error.descriptive_error import DescriptiveError
from models.response.edit_image_slide_response import EditImageSlide_Response

from routes.route_map import IMAGE_SLIDE

from flask import request

import pandas

@with_flask(IMAGE_SLIDE.EDIT.value, methods=["POST"])
def edit_image_slide():
    root_directory, report, image_slide = entities_for_editing_image_slide(request=request)
    parameter_name = get_or_panic(request.json, 'parameter_name', 'El nombre del parámetro a editar no está presente en la solicitud')
    parameter_value = get_or_panic(request.json, 'parameter_value', 'El valor del parámetro a editar no está presente en la solicitud')

    if parameter_name not in image_slide.parameters_dict:
        raise DescriptiveError(http_error_code=400, message='El nombre del parámetro no es válido para la diapostiva a editar')
    
    previous_value = image_slide.parameters_dict[parameter_name]
    image_slide.__setattr__(parameter_name, parameter_value)

    if previous_value != parameter_value:
        delete_preview(slide=image_slide)
        render_preview(root_directory=root_directory, slides=image_slide)

        image_slide.last_edit = pandas.Timestamp.now()

    save_slideshow(slideshow=report, root_directory=root_directory)

    return EditImageSlide_Response(preview=image_slide.preview).to_dict(), 200
